pathfinder/tests/test_blockchain_listener.py

```diff
@@ -0,0 +1,161 @@
+import pytest
+from web3 import Web3
+
+from pathfinder.utils.blockchain_listener import BlockchainListener, create_channel_event_topics
+
+
+def test_blockchain_listener(
+    web3: Web3,
+    wait_for_blocks,
+    generate_raiden_clients,
+    blockchain_listener: BlockchainListener,
+    ethereum_tester,
+):
+    blockchain_listener.required_confirmations = 4
+    blockchain_listener.start()
+    blockchain_listener.wait_sync()
+
+    unconfirmed_channel_open_events = []
+    confirmed_channel_open_events = []
+
+    blockchain_listener.add_unconfirmed_listener(
+        create_channel_event_topics(),
+        lambda e: unconfirmed_channel_open_events.append(e),
+    )
+    blockchain_listener.add_confirmed_listener(
+        create_channel_event_topics(),
+        lambda e: confirmed_channel_open_events.append(e),
+    )
+
+    # create unconfirmed channel
+    c1, c2 = generate_raiden_clients(2)
+    c1.open_channel(c2.address)
+
+    # the unconfirmed event should be available directly
+    wait_for_blocks(0)
+    assert len(unconfirmed_channel_open_events) == 1
+    assert unconfirmed_channel_open_events[0]['args']['participant1'] == c1.address
+    assert unconfirmed_channel_open_events[0]['args']['participant2'] == c2.address
+
+    # the confirmed event should be available after 4 more blocks as set above
+    wait_for_blocks(4)
+    assert len(confirmed_channel_open_events) == 1
+    assert confirmed_channel_open_events[0]['args']['participant1'] == c1.address
+    assert confirmed_channel_open_events[0]['args']['participant2'] == c2.address
+
+    blockchain_listener.stop()
+
+
+def test_blockchain_listener_nonexistant_contract(
+    web3: Web3,
+    wait_for_blocks,
+    generate_raiden_clients,
+    blockchain_listener: BlockchainListener,
+    ethereum_tester,
+):
+    blockchain_listener.required_confirmations = 4
+    blockchain_listener.contract_address = '0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa'
+    blockchain_listener.start()
+    blockchain_listener.wait_sync()
+
+    unconfirmed_channel_open_events = []
+    confirmed_channel_open_events = []
+
+    blockchain_listener.add_unconfirmed_listener(
+        create_channel_event_topics(),
+        lambda e: unconfirmed_channel_open_events.append(e),
+    )
+    blockchain_listener.add_confirmed_listener(
+        create_channel_event_topics(),
+        lambda e: confirmed_channel_open_events.append(e),
+    )
+
+    # create unconfirmed channel
+    c1, c2 = generate_raiden_clients(2)
+    c1.open_channel(c2.address)
+
+    # the unconfirmed event should be available directly
+    wait_for_blocks(0)
+    assert len(unconfirmed_channel_open_events) == 0
```

Comment:
The event should be available, so why does the assert expect 0? I would have expected 1.

Reply:
As the `BlockchainListener` listens to a different address we expect no output here.

But the comments are wrong, fixed that.